tests/basic_tests.cpp

```diff
@@ -17,6 +17,7 @@ TYPED_TEST(basic_tests, insane_url) {
   ASSERT_EQ(r->get_hostname(), "");
   ASSERT_EQ(r->get_port(), "");
   ASSERT_EQ(r->get_pathname(), "@EEEEEEEEEE");
+  ASSERT_FALSE(url->set_port("invalid80"));
```

Comment:
I recommend running tests before opening a PR

```suggestion
  ASSERT_FALSE(r->set_port("invalid80"));
```

Reply:
thank you I ran the tests but I got passed I wonder if I did the build wrong

I usedfirst this command: 
```
cmake -B build -DADA_TESTING=ON -DCMAKE_BUILD_TYPE=Debug                  
cmake --build build
```

after I used then this command for all tests:
`ctest --output-on-failure --test-dir build`

Reply:
> ctest --output-on-failure --test-dir build

What's the output of this command for you?

Reply:
I view this log: https://ctxt.io/2/AAB4CV4WFw

I put it on this site because it is too long